addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3862,11 +3873,21 @@ export class OdooEditor extends EventTarget {
                 setSelection(startContainer, startOffset, focusNode, focusOffset);
             }
         } else if (IS_KEYBOARD_EVENT_RIGHT_ARROW(ev)) {
+            const selection = this.document.getSelection();
+            const codeElement = closestElement(selection.anchorNode, 'code');
+            if (
+                codeElement?.classList.contains('o_inline_code') &&
+                selection.anchorOffset === nodeSize(selection.anchorNode) &&
+                !isZWS(codeElement?.nextSibling)
+            ) {
+                codeElement.after(document.createTextNode('\u200B'));
+                setSelection(codeElement.nextSibling, 1);
+            }
             if (ev.shiftKey) {
                 this._resetLinkZws();
             }
             getDeepRange(this.editable, { select: true });
-            let { anchorNode, anchorOffset, focusNode, focusOffset } = this.document.getSelection();
+            let { anchorNode, anchorOffset, focusNode, focusOffset } = selection;
             if (!focusNode) {
                 return;
             }
```

Comment:
Are you sure you shouldn't have put your logic _after_ this so you have a deep selection and you've checked that the selection is defined?

Reply:
Yes, you are right.i can try that as well. :+1: